File: src/dbflute/adf/meta/memberStatusdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type MemberStatusDbm_T struct {
	df.BaseDBMeta
	ColumnMemberStatusCode *df.ColumnInfo
	ColumnMemberStatusName *df.ColumnInfo
	ColumnDescription *df.ColumnInfo
	ColumnDisplayOrder *df.ColumnInfo
}

func (b *MemberStatusDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *MemberStatusDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var MemberStatusDbm *MemberStatusDbm_T

func Create_MemberStatusDbm() {
	MemberStatusDbm = new(MemberStatusDbm_T)
	MemberStatusDbm.TableDbName = "member_status"
	MemberStatusDbm.TableDispName = "member_status"
	MemberStatusDbm.TablePropertyName = "memberStatus"
	MemberStatusDbm.TableSqlName = new(df.TableSqlName)
	MemberStatusDbm.TableSqlName.TableSqlName = "member_status"
	MemberStatusDbm.TableSqlName.CorrespondingDbName = MemberStatusDbm.TableDbName

	var memberStatus df.DBMeta
	memberStatus = MemberStatusDbm
	MemberStatusDbm.DBMeta=&memberStatus
	memberStatusCodeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo member_status_code
	memberStatusCodeSqlName.ColumnSqlName = "member_status_code"
	memberStatusCodeSqlName.IrregularChar = false
	MemberStatusDbm.ColumnMemberStatusCode = df.CCI(&memberStatus, "member_status_code", memberStatusCodeSqlName, "", "", "String.class", "memberStatusCode", "", true, false,true, "bpchar", 3, 0, "",false,"","", "","memberList,memberLoginList","",false,"string")
	memberStatusNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo member_status_name
	memberStatusNameSqlName.ColumnSqlName = "member_status_name"
	memberStatusNameSqlName.IrregularChar = false
	MemberStatusDbm.ColumnMemberStatusName = df.CCI(&memberStatus, "member_status_name", memberStatusNameSqlName, "", "", "String.class", "memberStatusName", "", false, false,true, "varchar", 50, 0, "",false,"","", "","","",false,"string")
	descriptionSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo description
	descriptionSqlName.ColumnSqlName = "description"
	descriptionSqlName.IrregularChar = false
	MemberStatusDbm.ColumnDescription = df.CCI(&memberStatus, "description", descriptionSqlName, "", "", "String.class", "description", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	displayOrderSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo display_order
	displayOrderSqlName.ColumnSqlName = "display_order"
	displayOrderSqlName.IrregularChar = false
	MemberStatusDbm.ColumnDisplayOrder = df.CCI(&memberStatus, "display_order", displayOrderSqlName, "", "", "Integer.class", "displayOrder", "", false, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")

	MemberStatusDbm.ColumnInfoList = new(df.List)
	MemberStatusDbm.ColumnInfoList.Add(MemberStatusDbm.ColumnMemberStatusCode)
	MemberStatusDbm.ColumnInfoList.Add(MemberStatusDbm.ColumnMemberStatusName)
	MemberStatusDbm.ColumnInfoList.Add(MemberStatusDbm.ColumnDescription)
	MemberStatusDbm.ColumnInfoList.Add(MemberStatusDbm.ColumnDisplayOrder)


	MemberStatusDbm.ColumnInfoMap=make(map[string]int)
	MemberStatusDbm.ColumnInfoMap["memberStatusCode"]=0
		MemberStatusDbm.ColumnInfoMap["memberStatusName"]=1
		MemberStatusDbm.ColumnInfoMap["description"]=2
		MemberStatusDbm.ColumnInfoMap["displayOrder"]=3
	    MemberStatusDbm.PrimaryKey = true
    MemberStatusDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &memberStatus
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(MemberStatusDbm.ColumnMemberStatusCode)

	MemberStatusDbm.PrimaryInfo = new(df.PrimaryInfo)
	MemberStatusDbm.PrimaryInfo.UniqueInfo = ui
	
	var memberStatusMeta df.DBMeta = MemberStatusDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["MemberStatus"] = &memberStatusMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/MemberStatusDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of member_status. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class MemberStatusDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final MemberStatusDbm _instance = new MemberStatusDbm();
    private MemberStatusDbm() {}
    public static MemberStatusDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((MemberStatus)et).getMemberStatusCode(), (et, vl) -> ((MemberStatus)et).setMemberStatusCode((String)vl), "memberStatusCode");
        setupEpg(_epgMap, et -> ((MemberStatus)et).getMemberStatusName(), (et, vl) -> ((MemberStatus)et).setMemberStatusName((String)vl), "memberStatusName");
        setupEpg(_epgMap, et -> ((MemberStatus)et).getDescription(), (et, vl) -> ((MemberStatus)et).setDescription((String)vl), "description");
        setupEpg(_epgMap, et -> ((MemberStatus)et).getDisplayOrder(), (et, vl) -> ((MemberStatus)et).setDisplayOrder(cti(vl)), "displayOrder");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "member_status";
    protected final String _tableDispName = "member_status";
    protected final String _tablePropertyName = "memberStatus";
    protected final TableSqlName _tableSqlName = new TableSqlName("member_status", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnMemberStatusCode = cci("member_status_code", "member_status_code", null, null, String.class, "memberStatusCode", null, true, false, true, "bpchar", 3, 0, null, false, null, null, null, "memberList,memberLoginList", null, false);
//"string"
    protected final ColumnInfo _columnMemberStatusName = cci("member_status_name", "member_status_name", null, null, String.class, "memberStatusName", null, false, false, true, "varchar", 50, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnDescription = cci("description", "description", null, null, String.class, "description", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnDisplayOrder = cci("display_order", "display_order", null, null, Integer.class, "displayOrder", null, false, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"

    /**
     * member_status_code: {PK, NotNull, bpchar(3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberStatusCode() { return _columnMemberStatusCode; }
    /**
     * member_status_name: {NotNull, varchar(50)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberStatusName() { return _columnMemberStatusName; }
    /**
     * description: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDescription() { return _columnDescription; }
    /**
     * display_order: {UQ, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDisplayOrder() { return _columnDisplayOrder; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnMemberStatusCode());
        ls.add(columnMemberStatusName());
        ls.add(columnDescription());
        ls.add(columnDisplayOrder());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnMemberStatusCode()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() { return hpcui(columnDisplayOrder()); }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------
    /**
     * member by member_status_code, named 'memberList'.
     * @return The information object of referrer property. (NotNull)
     */
    public ReferrerInfo referrerMemberList() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnMemberStatusCode(), MemberDbm.getInstance().columnMemberStatusCode());
        return cri("fk_member_member_status", "memberList", this, MemberDbm.getInstance(), mp, false, "memberStatus");
    }
    /**
     * member_login by login_member_status_code, named 'memberLoginList'.
     * @return The information object of referrer property. (NotNull)
     */
    public ReferrerInfo referrerMemberLoginList() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnMemberStatusCode(), MemberLoginDbm.getInstance().columnLoginMemberStatusCode());
        return cri("fk_member_login_member_status", "memberLoginList", this, MemberLoginDbm.getInstance(), mp, false, "memberStatus");
    }

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.MemberStatus"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.MemberStatusCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.MemberStatusBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<MemberStatus> getEntityType() { return MemberStatus.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public MemberStatus newEntity() { return new MemberStatus(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((MemberStatus)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((MemberStatus)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
